project-api/project-api-test/src/test/java/io/ballerina/projects/test/OldPackageMigrationTests.java

```diff
@@ -118,8 +128,14 @@ public void testOldPackageWithLocalDependencies(ITestContext ctx) throws IOExcep
 
         DiagnosticResult diagnosticResult = compilation.diagnosticResult();
         diagnosticResult.diagnostics().forEach(OUT::println);
-        Assert.assertEquals(diagnosticResult.diagnosticCount(), 3, "Unexpected compilation diagnostics");
+        Assert.assertEquals(diagnosticResult.diagnosticCount(), 4, "Unexpected compilation diagnostics");
         Iterator<Diagnostic> diagnosticIterator = diagnosticResult.diagnostics().iterator();
+        // Check the older distribution version warning
+        Assert.assertEquals(diagnosticIterator.next().message(),
+                "Detected an attempt to compile this package using Swan Lake Update 5. " +
```

Comment:
The version will be `6` after the next version bump and the test will fail at that point. We must derive the version from the short version here.